packages/yarnpkg-fslib/tests/NodeFS.test.ts

```diff
@@ -0,0 +1,26 @@
+import {NodeFS}               from '../sources/NodeFS';
+import {xfs, ppath, Filename} from '../sources';
+
+const nodeFs = new NodeFS();
+
+describe(`NodeFS`, () => {
+  it(`should support PathBuffers`, async () => {
+    const tmpdir = await xfs.mktempPromise();
+
+    const pathBuffer = new ppath.PathBuffer(ppath.join(tmpdir, `file.txt` as Filename));
+
+    await nodeFs.writeFilePromise(pathBuffer, `...`);
+
+    await expect(nodeFs.readFilePromise(pathBuffer, `utf8`)).resolves.toStrictEqual(`...`);
+  });
+
+  it(`should support FileURLs`, async () => {
+    const tmpdir = await xfs.mktempPromise();
+
+    const url = new ppath.FileURL(ppath.join(tmpdir, `file.txt` as Filename));
+
+    await nodeFs.writeFilePromise(url, `...`);
+
+    await expect(nodeFs.readFilePromise(url, `utf8`)).resolves.toStrictEqual(`...`);
```

Comment:
Can you duplicate these checks to also check the regular path file content? Otherwise we check if URLs are deserialized into deterministic paths, but not whether they are deserialized into the right path.